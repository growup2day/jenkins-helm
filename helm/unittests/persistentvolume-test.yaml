templates:
  - persistentvolume.yaml
tests:
  - it: default manifest should match snapshot
    asserts:
      - matchSnapshot: {}
  - it: storage request should match snapshot
    set:
      persistentVolume:
        storage: 31Gi
    asserts:
      - equal:
          path: spec.resources.requests.storage
          value: 31Gi
  - it: single additional volume should match snapshot
    set:
      persistentVolume:
        storage: 7Gi
        additionalVolumes:
          - name: gitref-volume
            storage: 2Gi
            accessMode: ReadOnlyMany
    asserts:
      - hasDocuments:
          count: 2
      - matchSnapshot: {}
  - it: multiple additional volumes should match snapshot
    set:
      persistentVolume:
        storage: 7Gi
        additionalVolumes:
          - name: gitref-volume
            storage: 2Gi
            accessMode: ReadOnlyMany
          - name: maven-volume
            storage: 3Gi
            accessMode: ReadWriteMany
    asserts:
      - hasDocuments:
          count: 3
      - matchSnapshot: {}
  - it: "Should match snapshot when 'preventUninstall' is true for persistentVolume"
    set:
      persistentVolume:
        preventUninstall: true
    asserts:
      - matchSnapshot: {}
  - it: "Should match snapshot when 'preventUninstall' is true for one additional volume"
    set:
      persistentVolume:
        storage: 7Gi
        additionalVolumes:
          - name: gitref-volume
            storage: 2Gi
            accessMode: ReadOnlyMany
            preventUninstall: true
    asserts:
      - matchSnapshot: {}
  - it: "Should match snapshot when 'preventUninstall' is false for one additional volume"
    set:
      persistentVolume:
        storage: 7Gi
        additionalVolumes:
          - name: maven-volume
            storage: 3Gi
            accessMode: ReadWriteMany
            preventUninstall: false
    asserts:
      - matchSnapshot: {}
  - it: "Should match snapshot when 'preventUninstall' is set on multiple additional volumes"
    set:
      persistentVolume:
        storage: 7Gi
        additionalVolumes:
          - name: maven-volume
            storage: 3Gi
            accessMode: ReadWriteMany
            preventUninstall: true
          - name: gitref-volume
            storage: 3Gi
            accessMode: ReadWriteMany
    asserts:
      - matchSnapshot: {}