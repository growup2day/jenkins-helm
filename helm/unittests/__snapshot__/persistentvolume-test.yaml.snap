default manifest should match snapshot:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
Should match snapshot when 'preventUninstall' is false for one additional volume:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 7Gi
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins-maven-volume
    spec:
      accessModes:
      - ReadWriteMany
      resources:
        requests:
          storage: 3Gi
Should match snapshot when 'preventUninstall' is set on multiple additional volumes:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 7Gi
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      annotations:
        helm.sh/resource-policy: keep
      name: RELEASE-NAME-jenkins-maven-volume
    spec:
      accessModes:
      - ReadWriteMany
      resources:
        requests:
          storage: 3Gi
  3: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins-gitref-volume
    spec:
      accessModes:
      - ReadWriteMany
      resources:
        requests:
          storage: 3Gi
Should match snapshot when 'preventUninstall' is true for one additional volume:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 7Gi
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      annotations:
        helm.sh/resource-policy: keep
      name: RELEASE-NAME-jenkins-gitref-volume
    spec:
      accessModes:
      - ReadOnlyMany
      resources:
        requests:
          storage: 2Gi
Should match snapshot when 'preventUninstall' is true for persistentVolume:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      annotations:
        helm.sh/resource-policy: keep
      name: RELEASE-NAME-jenkins
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
default manifest should match snapshot:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
multiple additional volumes should match snapshot:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 7Gi
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins-gitref-volume
    spec:
      accessModes:
      - ReadOnlyMany
      resources:
        requests:
          storage: 2Gi
  3: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins-maven-volume
    spec:
      accessModes:
      - ReadWriteMany
      resources:
        requests:
          storage: 3Gi
single additional volume should match snapshot:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 7Gi
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-jenkins-gitref-volume
    spec:
      accessModes:
      - ReadOnlyMany
      resources:
        requests:
          storage: 2Gi
